Gems/Terrain/Code/Source/TerrainRenderer/TerrainDetailMaterialManager.cpp

```diff
@@ -758,61 +825,51 @@ namespace Terrain
+                            ++foundMaterials;
+                            secondWeight += surfaceTagWeight.m_weight;
+                            pixel.m_material2 = aznumeric_cast<uint8_t>(materialId);
+                        }
+                        else if (materialId == pixel.m_material2)
+                        {
+                            // Same material as the second material, just add the weights.
+                            secondWeight += surfaceTagWeight.m_weight;
                         }
                         else
                         {
-                            // Second material is valid, weight is relative based on first material's weight.
-                            pixels.at(index).m_material2 = aznumeric_cast<uint8_t>(materialId);
-                            float totalWeight = firstWeight + surfaceTagWeight.m_weight;
-                            float blendWeight = 1.0f - (firstWeight / totalWeight);
-                            pixels.at(index).m_blend = aznumeric_cast<uint8_t>(AZStd::round(blendWeight * 255.0f));
                             break;
                         }
                     }
-                    continue; // search for second material
                 }
-                else
-                {
-                    // No more valid materials in list since surfaceTagWeight is ordered.
-
-                    uint8_t defaultMaterial = region->m_defaultDetailMaterialId == InvalidDetailMaterailId ? m_passthroughMaterialId :
-                        aznumeric_cast<uint8_t>(m_detailMaterials.GetData(region->m_defaultDetailMaterialId).m_detailMaterialBufferIndex);
-
-                    if (isFirstMaterial)
-                    {
-                        // Only one material and it's the default material.
-                        pixels.at(index).m_material1 = defaultMaterial;
-                    }
-                    else
-                    {
-                        // Second material is default, weight is exactly what the first material requested
-                        pixels.at(index).m_material2 = defaultMaterial;
-                        float blendWeight = 1.0f - AZStd::clamp<float>(firstWeight, 0.0f, 1.0f);
-                        pixels.at(index).m_blend = aznumeric_cast<uint8_t>(AZStd::round(blendWeight * 255.0f));
-                    }
-                }
-
-                if (pixels.at(index).m_material1 == pixels.at(index).m_material2)
-                {
-                    // If the materials are the same, then make the blend 100% on the first id so the shader
-                    // doesn't blend identical materials
-                    pixels.at(index).m_blend = 0;
-                }
-
-                break;
             }
-            ++index;
+            
+            if (foundMaterials == 0)
+            {
+                // No materials found, so use the default materail.
```

Comment:
materail -> material.

(Sorry, no award yet for a PR without a single "materail" :P )

Reply:
At least the fact that I spelt it correctly once in the comment means I _know_ how to spell it ;)